src/Raven.Server/ServerWide/Commands/Subscriptions/PutSubscriptionCommand.cs

```diff
@@ -112,29 +113,39 @@ public override unsafe void Execute(TransactionOperationContext context, Table i
             }
         }
 
-        public long FindFreeId(TransactionOperationContext context, long subscriptionId)
+        public long FindFreeId(Dictionary<string, long> minimumSubscriptionIdPerDatabase, long subscriptionId, TransactionOperationContext context)
```

Comment:
This doesn't helps you for the _next_ `ExecutePutSubscriptionBatch` right?
Why not simply to get the last subscription id? and set the next id according to it?
e.g. if the last subId is `Foo.Bar` -> 'Foo.Bar/00000002`
and if it is  'Foo.Bar/00000002` ->  'Foo.Bar/00000003`

Reply:
It does. Next `ExecutePutSubscriptionBatch` is going to load the id from the storage.

Reply:
Maybe I miss something, but `minimumSubscriptionIdPerDatabase` is in-memory... so next time `FindFreeId` is called by a _different_ command you will be reading all the subscriptions, no?

Reply:
`FindFreeId` is only used for the `ExecutePutSubscriptionBatch`.

Reply:
I simplified the code and removed it from `PutSubscriptionCommand` to make it more clear